python/semantic_kernel/connectors/search/google/google_search.py

```diff
@@ -0,0 +1,194 @@
+            results=self._get_text_search_results(results),
+            total_count=self._get_total_count(results, options),
+            metadata=self._get_metadata(results),
+        )
+
+    async def get_search_results(
+        self, query: str, options: "SearchOptions | None" = None, **kwargs
+    ) -> "KernelSearchResults[GoogleSearchResult]":
+        """Search for text, returning a KernelSearchResult with the results directly from the service."""
+        options = self._get_options(options, **kwargs)
+        results = await self._inner_search(query=query, options=options)
+        return KernelSearchResults(
+            results=self._get_google_search_results(results),
+            total_count=self._get_total_count(results, options),
+            metadata=self._get_metadata(results),
+        )
+
+    async def _get_result_strings(self, response: GoogleSearchResponse) -> AsyncIterable[str]:
+        if response.items is None:
+            return
+        for item in response.items:
+            yield item.snippet or ""
+
+    async def _get_text_search_results(self, response: GoogleSearchResponse) -> AsyncIterable[TextSearchResult]:
+        if response.items is None:
+            return
+        for item in response.items:
+            yield TextSearchResult(
+                name=item.title,
+                value=item.snippet,
+                link=item.link,
+            )
+
+    async def _get_google_search_results(self, response: GoogleSearchResponse) -> AsyncIterable[GoogleSearchResult]:
+        if response.items is None:
+            return
+        for val in response.items:
+            yield val
+
+    def _get_metadata(self, response: GoogleSearchResponse) -> dict[str, Any]:
+        return {
+            "search_time": response.search_information.search_time if response.search_information else 0,
+        }
+
+    def _get_total_count(self, response: GoogleSearchResponse, options: TextSearchOptions) -> int | None:
+        total_results = (
+            None
+            if not options.include_total_count
+            else response.search_information.total_results or None
+            if response.search_information
+            else None
+        )
+        if total_results is not None:
+            return int(total_results)
+        return None
+
+    def _get_options(self, options: "SearchOptions | None", **kwargs: Any) -> TextSearchOptions:
+        if options is not None and isinstance(options, TextSearchOptions):
+            return options
+        try:
+            return TextSearchOptions(**kwargs)
+        except ValidationError:
+            return TextSearchOptions()
+
+    async def _inner_search(self, query: str, options: TextSearchOptions) -> GoogleSearchResponse:
+        self._validate_options(options)
+
+        logger.info(
+            f"Received request for google web search with \
+                params:\nquery: {query}\nnum_results: {options.top}\noffset: {options.skip}"
```

Comment:
The query may contain sensitive information, I'd expect to only log at trace or debug level. Is this OK to log this at info level in Python?

Reply:
Good point, removed both here and for Bing